File: Cpp/1694_Reformat_Phone_Number.cpp
class Solution {
public:
    string reformatNumber(string number) {
        string result;
        
        int count = 0;
        for(auto& ch : number) {
            if (ch < '0' || ch > '9') continue;
            result += ch;
            ++count;
            
            if(count % 3 == 0) result += '-';
        }
        
        if(result.back() == '-') {
            result.pop_back();
        } else if(result[result.size() - 2] == '-') {
            swap (result[result.size() - 2], result[result.size() - 3]);
        }
        
        return result;
    }
};

File: Kotlin/1694_Reformat_Phone_Number.kt
class Solution {
    fun reformatNumber(number: String): String {
        val result = StringBuilder()
        var count = 0
        
        for(ch in number) {
            if(!ch.isDigit()) continue
            
            result.append(ch)
            ++count
            if(count % 3 == 0) result.append('-')
        }

        if(result[result.length - 1] == '-') {
            result.deleteCharAt(result.length - 1)
        } else if(result[result.length - 2] == '-') {
            val ch1 = result[result.length - 2]
            val ch2 = result[result.length - 3]
            result.setCharAt(result.length - 2, ch2)
            result.setCharAt(result.length - 3, ch1)
        }
        
        return result.toString()
    }
}